flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/SlotManagerConfiguration.java

```diff
@@ -63,8 +63,10 @@ public SlotManagerConfiguration(
 		this.slotRequestTimeout = Preconditions.checkNotNull(slotRequestTimeout);
 		this.taskManagerTimeout = Preconditions.checkNotNull(taskManagerTimeout);
 		this.waitResultConsumedBeforeRelease = waitResultConsumedBeforeRelease;
-		this.slotMatchingStrategy = slotMatchingStrategy;
-		this.defaultWorkerResourceSpec = defaultWorkerResourceSpec;
+		this.slotMatchingStrategy = Preconditions.checkNotNull(slotMatchingStrategy);
+		this.defaultWorkerResourceSpec = Preconditions.checkNotNull(defaultWorkerResourceSpec);
+		Preconditions.checkState(numSlotsPerWorker > 0);
+		Preconditions.checkState(maxSlotNum > 0);
```

Comment:
It would be better to move the hotfix commit before the main commits if possible.